### PR TITLE
Blob-URL Widget

### PR DESCRIPTION
**Pull request by james-gc**
#### Data-URL takes too long, and hangs main-thread (specially on mobile browser)
Problem with the data-url was, it was taking too long on base64 conversion, in our project we faced it, and we resolved it by creating new field, and implemented blob-url.
we can upload it by later on submit event by :
`let blob = await fetch(url).then(r => r.blob());` 

If this is related to existing tickets, include links to them as well.

### Checklist

* [x] **I'm updating documentation**
  - [x] I've [checked the rendering](https://react-jsonschema-form.readthedocs.io/en/latest/#contributing) of the Markdown text I've added
* [x] **I'm adding or updating code**
  - [x] I've added and/or updated tests
  - [x] I've updated [docs](https://react-jsonschema-form.readthedocs.io/) if needed
  - [x] I've run `npm run cs-format` on my branch to conform my code to [prettier](https://github.com/prettier/prettier) coding style
* [x] **I'm adding a new feature**
  - [x] I've updated the playground with an example use of the feature

### CODE REVIEW

**Comment by james-gc** [comment]
this PR creates a blob-url rather than data-url on FileWidget.
Problem with the data-url was, it was taking too long on base64 conversion, in our project we faced it, and we resolved it by creating new field, and implemented blob-url.
we can upload it by later on submit event by :
`let blob = await fetch(url).then(r => r.blob());`

**Comment by james-gc** [comment]
Removed prettier options configuration from package.json.
Now it resides from .prettierrc.
Previously it conflicts with the vscode auto-formatter, by having `.prettierrc` we can use same config for vscode and precommit hook

**Comment by epicfaace** [comment]
@james-gc thanks for the PR. However, we can't implement this change as it is right now without breaking backwards compatibility (others may depend on this library storing files as a `data-url` by default).

Can you modify this PR perhaps to support an _additional_ format called `blob-url` which stores it as a blob-url?

**Comment by james-gc** [comment]
@epicfaace thanks for reviewing, it will do that.
and so on I also changed styling for `choose` file button, and now it can also show the file-preview

**Comment by james-gc** [comment]
@epicfaace updated all test cases for new widget `BlobFileWidget`

**Comment by james-gc** [comment]
@epicfaace documentation for `BlobFileWidget` also updated

**Comment by kevin24greylabs** [comment]
👍  looks good

File: .prettierrc [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (epicfaace) [inline .prettierrc]: Can you remove this prettier config changes? Would like to keep this PR self-contained to only have the blob url changes.

Comment (james-gc) [reply .prettierrc]: :+1: Sure, I will update it.

**Comment by james-gc** [comment]
@epicfaace Closing this PR

**Pull request closed without merging**